### DISCUSSION TITLE
pass props in Layout.js

### DISCUSSION THREAD

**Discussion by MahdiMotallebi** [vercel/next.js#40543] [Help] [answered]
Hello guys,
I have Header.js & Footer.js component and i use those in layout.js
then i use layout.js in _app.js for all page.

i wanna dynamic header.js and footer.js, how can i use getStaticProps and get data then pass to layout.js?

thx.💖

**Comment by anshuman-8** [comment]
You can try using Context API, and change state and pass data to whole site from footer.js or header.js.

**Reply by MahdiMotallebi** [reply]
> You can try using Context API, and change state and pass data to whole site from footer.js or header.js.

Thanks for your reply.

But which page i should fetch data?
_document.js & _app.js not support getStaticProps and getServerSide function.

**Comment by leerob** [comment, marked as answer]
**Update:** We’re excited to announce [Next.js 13](https://nextjs.org/13), which includes the new `app/` directory (beta) with support for nested layouts, colocation of data fetching, streaming, Server Components by default, and much more.

Inside the `app/` directory, there is a powerful new way to fetch data with React’s `use()` hook and the extended Web `fetch()` API. This allows you to colocate data fetching directly inside components for the most flexibility, including inside layouts.

You can leave feedback here: https://github.com/vercel/next.js/discussions/41745

**Reply by MahdiMotallebi** [reply]
Many thanks.💖💖
